From 10790964b095943f3dcb533c511095ee3b4f6350 Mon Sep 17 00:00:00 2001
From: James Rosewell <james@51degrees.com>
Date: Fri, 18 Sep 2015 17:13:47 +0100
Subject: [PATCH 22/28] MINOR: global: Added new fields for 51Degrees device
 detection

Added support for version 3.2 of 51Degrees C library.

Added fields to store HTTP header names important to device detection
other than User-Agent.

Included a pool of worksets for use with Pattern device detection.
---
 include/types/global.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/types/global.h b/include/types/global.h
index 2039e42..cb04074 100644
--- a/include/types/global.h
+++ b/include/types/global.h
@@ -191,8 +191,15 @@ struct global {
 		char property_separator;    /* the separator to use in the response for the values. this is taken from 51degrees-property-separator from config. */
 		struct list property_names; /* list of properties to load into the data set. this is taken from 51degrees-property-name-list from config. */
 		char *data_file_path;
+		int header_count; /* number of HTTP headers related to device detection. */
+		struct chunk *header_names; /* array of HTTP header names. */
 #ifdef FIFTYONEDEGREES_H_PATTERN_INCLUDED
 		fiftyoneDegreesDataSet data_set; /* data set used with the pattern detection method. */
+		fiftyoneDegreesWorksetPool *pool; /* pool of worksets to avoid creating a new one for each request. */
+#endif
+#ifdef FIFTYONEDEGREES_H_TRIE_INCLUDED
+		int32_t *header_offsets; /* offsets to the HTTP header name string. */
+		fiftyoneDegreesDeviceOffsets device_offsets; /* Memory used for device offsets. */
 #endif
 		int cache_size;
 	} _51degrees;
-- 
2.4.9

